File: WebApplication2/Models/Invoice.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace FlatExpenses.Models
{
    public class Invoice
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        public string User { get; set; }

        public float Amount { get; set; }

        public DateTime Date { get; set; }

        public DateTime SaveDate { get; set; }
    }
}

File: Frontend/src/app/shared/invoice-api/Invoice.ts
export class Invoice {
    public id: AAGUID;
    public date: Date;
    public user: string;
    public amount: number;

    constructor(id: AAGUID, date: Date, user: string, amount: number, saveDate: Date){
        this.id = id;
        this.date = date;
        this.user = user;
        this.amount = amount;
    }
}
